File: leetcode2025/2025_10/leetcode2025-10-24.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
private:
    bool is_balanced(int x){
        vector<int> cnt(10, 0);
        while (x){
            if (cnt[x%10] >= x%10){
                return false;
            }
            cnt[x%10]++;
            x /= 10;
        }
        for (int i=0; i<10; i++){
            if (cnt[i] > 0 && cnt[i] != i){
                return false;
            }
        }
        return true;
    }
public:
    int nextBeautifulNumber(int n) {
        int res = n+1;
        while (!is_balanced(res)){
            res++;
        }
        return res;
    }
};

int main(){
    Solution s;
    int n = 1;
    cout << s.nextBeautifulNumber(n) << endl;
}

File: leetcode2025/2025_10/leetcode2025-10-24.py
class Solution:
    def nextBeautifulNumber(self, n: int) -> int:
        def is_balanced(x):
            # 判断x是否平衡
            cnt = [0]*10
            while x:
                if cnt[x%10] >= x%10:
                    return False
                cnt[x%10] += 1
                x //= 10
            for i in range(10):
                if cnt[i] > 0 and cnt[i] != i:
                    return False
            return True
        
        res = n+1
        while not is_balanced(res):
            res += 1
        return res

if __name__ == "__main__":
    s = Solution()
    n = 1
    print(s.nextBeautifulNumber(n))
